server/api.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/plugin"
+)
+
+// ServeHTTP handles HTTP requests to the plugin.
+func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
+	config := p.getConfiguration()
+
+	if err := config.IsValid(); err != nil {
+		http.Error(w, "This plugin is not configured.", http.StatusNotImplemented)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	switch path := r.URL.Path; path {
+	case "/webhook":
+		p.handleWebhook(w, r)
+	case "/profile.png":
+		p.handleProfileImage(w, r)
+	case "/api/v1/userinstalls":
+		p.handleUserInstalls(w, r)
+	default:
+		http.NotFound(w, r)
+	}
+}
+
+func (p *Plugin) handleUserInstalls(w http.ResponseWriter, r *http.Request) {
+	p.API.LogWarn(fmt.Sprintf("%+v", r.Header))
```

Comment:
Debug message?

Reply:
Yep, thanks for catching that. I thought I had cleaned them all up.